proxy/src/control/destination/mod.rs

```diff
@@ -23,22 +50,40 @@ pub struct Resolver {
     request_tx: mpsc::UnboundedSender<ResolveRequest>,
 }
 
+/// Requests that resolution updaes for `authority` be sent on `responder`.
 #[derive(Debug)]
 struct ResolveRequest {
     authority: DnsNameAndPort,
+    responder: Responder,
+}
+
+/// A handle through which response updates may be sent.
+#[derive(Debug)]
+struct Responder {
+    /// Sends updates from the controller to a `Resoliution`.
     update_tx: mpsc::UnboundedSender<Update>,
+
+    /// Indicates whether the corresponding `Resolution` is still active.
+    active: Arc<()>,
 }
 
 /// A `tower_discover::Discover`, given to a `tower_balance::Balance`.
 #[derive(Debug)]
 pub struct Resolution<B> {
+    /// Receives updates from the controller.
     update_rx: mpsc::UnboundedReceiver<Update>,
+
+    /// Allows `Responder` to detect when its `Resolution` has been lost.
```

Comment:
perhaps this comment should explain _how_ this works? if the reader doesn't know what happens when you drop a `Weak` reference, it looks a little magical...